### ISSUE TITLE
Merging PDFs using `--pages` removes checkbox/radio fillable field functionality

### ISSUE THREAD

**Issue by bsbeeks** [qpdf/qpdf#340]
When combining separate PDFs, the resulting PDF does not maintain some checkboxes and radio buttons. When inspecting the combined pdf in Adobe Pro DC, I noticed that the fields are still there, but they are no longer defined as radio/checkbox fields.

The interesting thing is that it doesn't do this for all checkbox/radio fields. However, I haven't been able to pin point what the difference is between the checkbox/radio fields that work and the ones that don't. 🤔

PDF field before combining:
<img width="952" alt="Screen Shot 2019-07-01 at 9 56 41 AM" src="https://user-images.githubusercontent.com/714084/60450246-a8e6eb80-9be6-11e9-834b-859d029f5330.png">

PDF field after combining:
<img width="920" alt="Screen Shot 2019-07-01 at 9 56 53 AM" src="https://user-images.githubusercontent.com/714084/60450263-b308ea00-9be6-11e9-9abc-1fa9361fcd25.png">

Here is the command I'm using with v8.4.0:
`qpdf --empty --pages 1.pdf 1-z 2.pdf -- combined.pdf`

I've attempted to use the following flags in various combinations with no success:
- `--preserve-unreferenced-resources`
- `--decode-level=none`
- `--normalize-content=n`
- `--preserve-unreferenced`

Here is combined PDF that has working and non-working checkboxes. Look at the 3rd page under the occupation section.
- https://cdn.ixn.tech/forms/d398c605-d07e-4d63-ada6-25dec7e21b6b.pdf

Here are the original PDFs that were used to create the above combined PDF:
- https://ixn-data.s3-us-west-2.amazonaws.com/forms/49/3d1c5830-3114-4d55-875f-3e64f0ce1992.pdf
- https://ixn-data.s3-us-west-2.amazonaws.com/forms/49/81009f23-780d-43f4-9c83-3903b7f1730b.pdf

**Comment by bsbeeks** [comment]
Been doing more digging and it appears as though it might be related to how the fillable fields are named. In the example files above, the fields that don't transfer over properly, all have the same name...and because of this it appears that somewhere down the line their names were appended with numbers to differentiate them. For example, there are 5 fields that appear to have been named "Life1_occupation_status". Their actual names are "Life1_occupation_status#0", "Life1_occupation_status#1", and so on. If I edit those field names and replace the "#" with "_", then they are transferred over properly when I merge the pdfs.

There are some others cases where the field name does not include a "#", and they are still not transferring over. So far, it appears that in those cases they are radio buttons that are "grouped", and therefore their names are the same. If I remove the fields from their group, then they transfer over properly.

So, to summarize it appears that:

1. Fields whose names include a "#" are not transferred over properly when merging.
2. Fields that are part of a group are not transferred over properly when merging.

**Comment by bsbeeks** [comment]
Still having this issue even after updating to 10.1.0. Unfortunately going to have to move to pdftk, as combining pdfs using that library doesn't result in a loss of fillable functionality.

**Comment by jberkenbilt** [comment]
@bsbeeks Sorry, it seems I overlooked this issue and never responded to it. This is a known issue and the most frequent feature request I get, and it is definitely on my roadmap to fix, but it is complicated. The work I'm doing now for 10.2 takes me a long way toward implementing it. It won't be in 10.2, but I'm hoping to get to it soonish, with the caveat that I get a surge of time to work on qpdf only periodically. I don't have a timeline. I will comment on this issue when I get to that functionality and/or tag you on which ever of the other issues ends up being the one to track that feature request. When I'm ready to look at this in depth, I will decide which issue to track it on. This is a good one because of the examples and the details. Stay tuned for more information. 

I apologize for overlooking it. If you report an issue in the future and I don't respond within a week or so, feel free to nudge. I usually try to respond promptly to issues but sometimes life happens and I overlook things.

**Comment by jberkenbilt** [comment]
FYI I assign the `pages` label to all issues that are related to the fact that qpdf doesn't yet manage document-level metadata (other than page labels) when splittng and merging files. 10.2 will have support for handling file attachments via a `--copy-attachments-from` parameter, and the infrastructure to support that has heavy overlap with form fields. For fields will probably be next up, though I may do tags first if they turn out to be significantly easier.

**Comment by jberkenbilt** [comment]
This will be fixed for 10.2
